examples/step-72/step-72.cc

```diff
@@ -0,0 +1,1147 @@
+      // over all quadrature points and degrees-of-freedom.
+      std::vector<Tensor<1, dim>> old_solution_gradients(
+        fe_values.n_quadrature_points);
+      fe_values.get_function_gradients(current_solution,
+                                       old_solution_gradients);
+
+      for (const unsigned int q : fe_values.quadrature_point_indices())
+        {
+          const double coeff =
+            1.0 / std::sqrt(1.0 + old_solution_gradients[q] *
+                                    old_solution_gradients[q]);
+
+          for (const unsigned int i : fe_values.dof_indices())
+            {
+              for (const unsigned int j : fe_values.dof_indices())
+                cell_matrix(i, j) +=
+                  (((fe_values.shape_grad(i, q)      // ((\nabla \phi_i
+                     * coeff                         //   * a_n
+                     * fe_values.shape_grad(j, q))   //   * \nabla \phi_j)
+                    -                                //  -
+                    (fe_values.shape_grad(i, q)      //  (\nabla \phi_i
+                     * coeff * coeff * coeff         //   * a_n^3
+                     * (fe_values.shape_grad(j, q)   //   * (\nabla \phi_j
+                        * old_solution_gradients[q]) //      * \nabla u_n)
+                     * old_solution_gradients[q]))   //   * \nabla u_n)))
+                   * fe_values.JxW(q));              // * dx
+
+              cell_rhs(i) -= (fe_values.shape_grad(i, q)  // \nabla \phi_i
+                              * coeff                     // * a_n
+                              * old_solution_gradients[q] // * u_n
+                              * fe_values.JxW(q));        // * dx
+            }
+        }
+    };
+
+    // The second lambda function that MeshWorker::mesh_loop() requires is
+    // one that performs the task of accumulating the local contributions
+    // in the global linear system. That is precisely what this one does,
+    // and the details of the implementation have been seen before. The
+    // primary point to recognize is that the local contributions are stored
+    // in the `copy_data` instance that is passed into this function. This
+    // `copy_data` has been filled with data during @a some call to the
+    // `cell_worker`.
+    auto copier = [dofs_per_cell, this](const CopyData &copy_data) {
+      const FullMatrix<double> &cell_matrix = copy_data.matrices[0];
+      const Vector<double> &    cell_rhs    = copy_data.vectors[0];
+      const std::vector<types::global_dof_index> &local_dof_indices =
+        copy_data.local_dof_indices[0];
+
+      for (unsigned int i = 0; i < dofs_per_cell; ++i)
+        {
+          for (unsigned int j = 0; j < dofs_per_cell; ++j)
+            system_matrix.add(local_dof_indices[i],
+                              local_dof_indices[j],
+                              cell_matrix(i, j));
+
+          system_rhs(local_dof_indices[i]) += cell_rhs(i);
```

Comment:
In my opinion, this is something we should only show once (step 3) and then always use `AffineConstraints`.

Reply:
I agree, and I even thought about making the change (see my [TODO list](https://github.com/dealii/dealii/pull/10393#issue-424080398)). We'd basically need to make the change consistently across step-15, step-72 and step-77, so I ultimately decided to leave this to another time. For the sake of consistency, I'd thus propose that we leave it how it is and make the improvement after the release.

Reply:
I agree -- consistency is useful!